moto/ec2/utils.py

```diff
@@ -675,3 +675,22 @@ def filter_iam_instance_profiles(iam_instance_profile_arn, iam_instance_profile_
             instance_profile = None
 
     return instance_profile
+
+
+def describe_tag_filter(filters, instances):
+    instances = list(instances)
+    for key in filters:
+        if key.startswith('tag:'):
+            match = re.match(r'tag:(.*)', key)
+            if match:
+                tag_key_name = match.group(1)
+            for instance in instances:
+                need_delete = True
+                for tag in instance.get_tags():
```

Comment:
Not sure I fully understand the logic in lines 688-695. Assuming `get_tags()` returns a list of two tags, say, `[tag1, tag2]`, and if the value of `tag2` matches, whereas the value of `tag1` does **not** match, then the `instance` would NOT get deleted. Is this the intended behavior here? (I would have expected that the instance should not be part of the result list if one of the tags is not matching)